spec/ndx-ophys-devices.extensions.yaml

```diff
@@ -1,8 +1,216 @@
+- neurodata_type_def: Indicator
+  neurodata_type_inc: Device
+  doc: Extends Device to hold metadata on the Indicator.
   attributes:
-  - name: trode_id
-    dtype: int32
-    doc: The tetrode ID.
+  - name: label
+    dtype: text
+    doc: Indicator standard notation.
+  - name: injection_location
+    dtype: text
+    doc: Injection brain region name.
+    required: false
+  - name: injection_coordinates_in_mm
+    dtype: float
+    shape:
+    - 3
+    doc: Indicator injection location in stereotactic coordinates (AP, ML, DV) mm
+      relative to Bregma.
+    required: false
+- neurodata_type_def: OpticalFiber
+  neurodata_type_inc: Device
+  doc: Extends Device to hold metadata on the Optical Fiber.
+  attributes:
+  - name: model
+    dtype: text
+    doc: Model of optical fiber.
+    required: false
+  - name: numerical_aperture
+    dtype: float
+    doc: Numerical aperture, e.g., 0.39 NA.
+    required: false
+  - name: core_diameter_in_um
+    dtype: float
+    doc: Core diameter in micrometers.
+    required: false
+- neurodata_type_def: ExcitationSource
+  neurodata_type_inc: Device
+  doc: Extends Device to hold metadata on the Excitation Source.
+  attributes:
+  - name: model
+    dtype: text
+    doc: Model of excitation source device.
+    required: false
+  - name: illumination_type
+    dtype: text
+    doc: 'Type of illumination. Suggested values: LED, LAser or more specifically
+      Gas, Metal vapor, Solid state, Dye, Semiconductor, Free electron, etc.'
+  - name: excitation_wavelength_in_nm
+    dtype: float
+    doc: Excitation wavelength of the stimulation light (nanometers).
+  - name: peak_power_in_W
+    dtype: float
+    doc: Incident power of stimulation device (in Watts).
+    required: false
+  - name: peak_pulse_energy_in_J
+    dtype: float
+    doc: If device is pulsed light source, pulse energy (in Joules).
+    required: false
+  - name: intensity_in_W_per_m2
+    dtype: float
+    doc: Intensity of the excitation in W/m^2, if known.
+    required: false
+  - name: pulse_rate_in_Hz
+    dtype: float
+    doc: If device is pulsed light source, pulse rate (in Hz) used for stimulation.
+    required: false
+  - name: exposure_time_in_s
+    dtype: float
+    doc: Exposure time of the sample (in sec).
+    required: false
+- neurodata_type_def: Photodetector
+  neurodata_type_inc: Device
+  doc: Extends Device to hold metadata on the Photodetector.
+  attributes:
+  - name: model
+    dtype: text
+    doc: Model of photodetector device.
+    required: false
+  - name: detector_type
+    dtype: text
+    doc: 'Technology used to detect the light. Suggested values: CCD, Intensified
+      CCD, Analog video, PMT, Photodiode, CMOS, EBCCD, FTIR.'
+  - name: detected_wavelength_in_nm
+    dtype: float
+    doc: Wavelength detected by photodetector.
+  - name: gain
+    dtype: float
+    doc: Gain on the photodetector.
+- neurodata_type_def: DichroicMirror
```

Comment:
First time seeing this one - how do you envision this fitting into ophys metadata? Just one extra thing on the outside that doesn't link to any series?

Reply:
The dichroic mirrors are basically another type of optical filter that has the specific function of separating the emitted light from the excited light. It's another component of a general microscopy setup.

Reply:
Yes I've heard of their role in the experiment setup - what I mean is, in terms of the `ndx-microscopy` schema, how do you envision them fitting in?

Reply:
A subtype of optical filter? And in-between link between a microscope and an optical filter? Something else entirely?

Reply:
In the ndx-microscopy I proposed to adopt the same approach as in the ndx-fiber-photometry, where we have a table (`MicroscopyTable`) where each row defines a specific configuration of the microscopy setup.

> Yes I've heard of their role in the experiment setup - what I mean is, in terms of the ndx-microscopy schema, how do you envision them fitting in?

In the `MicroscopyTable`, the `DichroicMirror` will be referenced in a dedicated column as part of the microscopy setup, along with the other "pieces": emission filter, excitation filter, excitation source, objective lens, photodetector, etc.

Then, in the `MicroscopySeries`, we will have a `microscopy_table_region` field where we can reference a row of the `MicroscopyTable` corresponding to the specific configuration of the microscope used to record the imaging data. In this way, we replace the OpticalChannel concept and have all the information in one place.

Reply:
TODO: remove Microscope from this extension

MicroscopeConfiguration will live in ndx-microscopy